.gitmodules

```diff
@@ -1,36 +1,100 @@
 [submodule "libsodium/libsodium"]
 	path = libsodium/libsodium
 	url = https://github.com/jedisct1/libsodium.git
+	# libsodium doesn't tag new releases since 0.18.0, 'stable' branch is considered
+	# to be the latest stable release. See:
+	# https://github.com/jedisct1/libsodium/issues/1041#issuecomment-799812417
+	# However since idf_component.yml had the version already set to 0.20.0,
+	# we can't update the version automatically (it would be set it back to 0.18.0~n)
+	# autoupdate = false
 [submodule "cbor/tinycbor"]
 	path = cbor/tinycbor
 	url = https://github.com/intel/tinycbor.git
+	autoupdate = true
+	autoupdate-branch = main
+	autoupdate-tag-glob = v[0-9]*.[0-9]*.[0-9]*
+	autoupdate-manifest = cbor/idf_component.yml
+	autoupdate-ver-regex = v([0-9]+).([0-9]+).([0-9]+)
```

Comment:
Should we consider keeping some default values for branch, tag-glob and ver-regex fields? If we take most common syntax (e.g., version as `vx.y.z`) then we can reduce the additional fields specified for all external submodules.

Reply:
I kept these explicit in order to force the component maintainer to go and check the tag naming pattern the repository actually uses.

Alternatively I could make the action fail if not a single tag with the given naming pattern is found. In that case I'll also make the action run when .gitmodules file is changed, with a --dry-run argument.